### PR TITLE
Make the Keypather class auto-instance.

### PR DESCRIPTION
**Pull request by justsml**
Hey @tjmehta I love your code's approach & scope of feature support... Anyway I will create a separate  issue to discuss some of my thoughts. Anyway, here's a patch to eliminate the need for double parenthesis:
See my updates to the README.md

### CODE REVIEW

**Comment by justsml** [comment]
Hey I wasn't sure how I broke it... I thought the other option 'force' tests would fail if I broke the option...

I'm trying to detect options if supplied when explicitly calling into the Keypather(opts) class, 
Regardless of whether or not options were passed, I wanted to try an auto-instance pattern, something like this: ```if (!(this instanceof Keypather)) return new Keypather(opts)``` in the index.js. Hopefully eliminating the double sets of parens needed.

File: index.js [hunk 1]
```diff
@@ -1,10 +1,8 @@
...
-  var keypather = new Keypather(opts && opts.force);
-  return keypather;
-};
+module.exports = Keypather;
 
-function Keypather (force) {
-  this.force = (force !== undefined) ? Boolean(force) : true; // force - default: true
+function Keypather (opts) {
+  if (!(this instanceof Keypather)) { return new Keypather(opts); }// Auto instantiate
+  this.force = (opts && opts.force && Boolean(opts.force) === false ? false : true); // force - default: true
```
Comment (stoeffel) [inline index.js]: This should be:
`this.force = (opts && typeof opts.force !== 'undefined' && Boolean(opts.force) === false ? false : true);`
else it's always true.

**Comment by stoeffel** [comment]
This wouldn't work since you have to create a instance of `keypather` somewhere. 

```js
var keypath = require('keypather');

// keypath is a function

 var obj = {
  ...
 };
 keypath.get(obj, "foo.bar.baz"); // This would break get is not a function of keypath.
```

**Comment by justsml** [comment]
Hey @stoeffel - Hey thanks for the reply.
What if the Keypather function itself was 'decorated' like this: (is that the term?)
```javascript
// Instead of setting the method on the prototype, something like this:
Keypather.get = function() { /* ... */ }
```
I'm not 100% sure, but is this a valid pattern to allow this kind of usage:
```Keypather.prototype.get = Keypather.get = function() { /* ... */ }```

**Comment by tjmehta** [comment]
Hmm, maybe this:

```js
module.exports = factoryAndInstance;
function factoryAndInstance (opts) {
  var keypather = new Keypather(opts && opts.force);
  return keypather;
};
Keypather.call(factoryAndInstance);
factoryAndInstance.__proto__ = new Keypather().__proto__;
```

**Comment by tjmehta** [comment]
^ that would export a functioning keypather instance defaulted with `force:true`

**Comment by justsml** [comment]
Hey guys... I updated the code... It's still failing... I was wondering how important having the force option is? It adds a layer of branching in the code that might be handled a little differently... I was thinking of moving this kind of functionality under something like ```keypath.trySet()``` otherwise you have to choose up front how every keypather call will be affected. ... @tjmehta @stoeffel any thoughts? ... Sorry if this is kinda drifting off topic...

File: index.js [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
   return keypather;
 };
+Keypather.call(factoryAndInstance);
+factoryAndInstance.__proto__ = new Keypather().__proto__;
 
-function Keypather (force) {
-  this.force = (force !== undefined) ? Boolean(force) : true; // force - default: true
+function Keypather (opts) {
+  if (!(this instanceof Keypather)) { return new Keypather(opts); }// Auto instantiate
+  this.force = (opts && typeof opts.force !== 'undefined' && Boolean(opts.force) === false ? false : true);
```
Comment (tjmehta) [inline index.js]: I don't think you need these modifications just revert it to what it was
`this.force = (force !== undefined) ? Boolean(force) : true; // force - default: true`

**Comment by justsml** [comment]
Just updated... Woo hoo!
Thanks guys

**Comment by stoeffel** [comment]
Travis is still failing. You should run the test before pushing (`npm test`).
And you should update the tests to check if it works without instantiating keypather.

**Comment by justsml** [comment]
Thanks @stoeffel - I was loading broken stuff for yours & @tjmehta 's feedback, sorry if it should have taken place in an Issue or wiki or whatever... 
I'll get time this weekend to load a bunch of updates/ideas...

**Comment by tjmehta** [comment]
Going to close this for now, feel free to reopen if you continue work on this.

**Pull request closed without merging**